listenbrainz_spark/stats_writer/stats_writer.py

```diff
@@ -0,0 +1,75 @@
+import pika
+import logging
+from listenbrainz_spark import config
+import time
+import json
+import sys
+
+class StatsWriter():
+    def __init__(self):
+        self.unique_ch = None
+        self.ERROR_RETRY_DELAY = 3
+
+    def connect_to_rabbitmq(self):
+        """Creates a RabbitMQ connection object
+        """
+        username = config.RABBITMQ_USERNAME
+        password = config.RABBITMQ_PASSWORD
+        host = config.RABBITMQ_HOST
+        port = config.RABBITMQ_PORT
+        virtual_host = config.RABBITMQ_VHOST
+
+        while True:
+            try:
+                credentials = pika.PlainCredentials(username, password)
+                connection_parameters = pika.ConnectionParameters(
+                    host=host,
+                    port=port,
+                    virtual_host=virtual_host,
+                    credentials=credentials,
+                )
+                self.connection =  pika.BlockingConnection(connection_parameters)
```

Comment:
Here too, we can find more specific errors to catch. See the LB main code for examples.